tests/modules/customauth.c

```diff
@@ -0,0 +1,179 @@
+#include "redismodule.h"
```

Comment:
Repost: I'm not convinced we need to build two modules, and I'm also not convinced these need to be separate from the auth.c file.

Reply:
Reposting an earlier comment. Perhaps we could use startup arguments so we can load the same module twice but have it register slightly different values.

Reply:
Having multiple modules loaded adds a bit more test coverage as we have some marker concept to iterate over all the modules processed. 

If we can load the same module twice with different names, that should also work.

Reply:
We will continue to have the second module for testing until there is an API / mechanism to properly support loading the same module (.so file) twice. Right now we can load the module twice, however, we do not have a way to allocate internal data and access it later on through the RMContext. If we try storing info in a pointer in the context of Module 1, the pointer gets overwritten by module  2. If we have a mechanism to hold info for a Module in the engine through the RMContext and access it later, this option can be used and the second testing module can be removed.

Reply:
The second point was that most of this should just be in auth.c is still valid, let's not just add more modules for very specific functionality.